app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java

```diff
@@ -28,29 +28,52 @@
 import org.springframework.context.annotation.Configuration;
 
 import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Creates and configures the main datastore
  */
 @Configuration
 public class DataStoreConfiguration {
     private static final Logger LOG = LoggerFactory.getLogger(SchemaCheck.class);
+    private static final Index FAKE_INDEX = new Index(null, null);
+
+    /**
+     * This basically ensures that we get at least one Index created in the system so
+     * that the index list injection in the jsonDB does not fail due to no Indexes
+     * being defined.
+     *
+     * @return
+     */
+    @Bean
+    public Index fakeIndex() {
+        return FAKE_INDEX;
+    }
 
     @Bean
     @Autowired
     @SuppressWarnings("PMD.EmptyCatchBlock")
-    public SqlJsonDB realTimeDB(DBI dbi) {
+    public SqlJsonDB jsonDB(DBI dbi, List<Index> beanIndexes) {
```

Comment:
I think we could use `Optional<List<Index>>` or `@Autowired(required = false)` here